deployer/cloudflaredeployer.go

```diff
@@ -0,0 +1,312 @@
+		}
+	}
+
+	for _, meta := range appSource.Applications() {
+		log.LogStart(fmt.Sprintf("deploying %s (%s) runnables as workers", meta.Identifier, job.domain))
+
+		runnablesByName := getRunnablesByName(appSource, meta)
+
+		workersToRoute := make(map[string]cfWorker)
+
+		for _, handler := range appSource.Handlers(meta.Identifier, meta.AppVersion) {
+			if handler.Input.Type != "request" {
+				log.LogWarn(fmt.Sprintf("unsupported handler type: %s; ignoring.", handler.Input.Type))
+				continue
+			}
+
+			if len(handler.Steps) != 1 {
+				log.LogWarn(fmt.Sprintf("only one step is currently supported per handler"))
+				continue
+			}
+
+			for _, step := range handler.Steps {
+				runnable, ok := runnablesByName[step.CallableFn.Fn]
+				if !ok {
+					return errors.Errorf("unexpected runnable %s", step.CallableFn.Fn)
+				}
+
+				script, err := renderRequestWorker(templatesPath, handler.Input.Method, handler.Input.Resource)
+				if err != nil {
+					return errors.Wrap(err, "failed to render request worker")
+				}
+				scriptName := fmt.Sprintf("%s-request-%s", strings.ReplaceAll(meta.Identifier, ".", "-"), runnable.Name)
+				if err := deployRunnable(api, &runnable, scriptName, script, job.domain); err != nil {
+					return errors.Wrap(err, "failed to deploy runnable")
+				}
+
+				path := handler.Input.Resource
+				workersToRoute[scriptName] = cfWorker{
+					Path:           path,
+					Method:         handler.Input.Method,
+					ServiceBinding: toWorkerBinding(scriptName),
+				}
+
+				log.LogInfo(fmt.Sprintf("deployed runnable %s", runnable.Name))
+			}
+		}
+
+		if err := deployRouter(templatesPath, api, workersToRoute, job.domain); err != nil {
+			return errors.Wrap(err, "failed to create router")
+		}
+		routerRoute := fmt.Sprintf("%s/*", job.domain)
+		if err := createRoute(api, routerRoute, routerScriptName, job.domain); err != nil {
+			log.LogWarn(fmt.Sprintf("error while creating route: %s", err))
+		}
+
+		log.LogDone(fmt.Sprintf("deployed to Cloudflare at %s", routerRoute))
+	}
+
+	return nil
+}
+
+func getAPIClient() (*cloudflare.API, error) {
+	apiToken := os.Getenv("CF_API_TOKEN")
+	accountId := os.Getenv("CF_ACCOUNT_ID")
+
+	if apiToken == "" || accountId == "" {
+		return nil, errors.New("missing CF_API_TOKEN or CF_ACCOUNT_ID")
+	}
+
+	api, err := cloudflare.NewWithAPIToken(apiToken, cloudflare.UsingAccount(accountId))
+	if err != nil {
+		return nil, err
+	}
+
+	return api, nil
+}
+
+func getRunnablesByName(appSource appsource.AppSource, meta appsource.Meta) map[string]directive.Runnable {
+	runnablesByName := make(map[string]directive.Runnable)
+	runnables := appSource.Runnables(meta.Identifier, meta.AppVersion)
+
+	for _, runnable := range runnables {
+		runnablesByName[runnable.Name] = runnable
+	}
+
+	return runnablesByName
+}
+
+func renderRouterWorker(templatesPath string, workersToRoute map[string]cfWorker) (string, error) {
+	templatePath := path.Join(templatesPath, "cf/router_worker.js.tmpl")
```

Comment:
You can pass multiple parts into `Join`, i.e. `(templatesPath, "cf", "router_worker.js.tmpl")`... this ensures it will work on all platforms.